srv/processors-service.js

```diff
@@ -3,28 +3,19 @@ const cds = require('@sap/cds')
 class ProcessorsService extends cds.ApplicationService {
   /** Registering custom event handlers */
   init() {
-    this.before("UPDATE", "Incidents", (req) => this.onUpdate(req));
-    this.after("READ", "Incidents", (data) => this.changeUrgencyDueToSubject(data));
-
-    return super.init();
+    this.before("SAVE", "Incidents", (req) => this.ensureTimestampIsPresent(req))
+    return super.init()
   }
 
-  changeUrgencyDueToSubject(data) {
-    if (data) {
-      const incidents = Array.isArray(data) ? data : [data];
-      incidents.forEach((incident) => {
-        if (incident.title?.toLowerCase().includes("urgent")) {
-          incident.urgency = { code: "H", descr: "high" };
+  async ensureTimestampIsPresent (req) {
+    if (req.data.conversations) {
+      for(const conversation of req.data.conversations) {
+        if(!conversation.timestamp) {
+            conversation.timestamp = new Date(Date.now())
```

Comment:
Although we initially discussed data validation I could not come up with a validation that was a) simple enough but b) not already covered by [existing annotations](https://cap.cloud.sap/docs/guides/providing-services#mandatory). So, this code fills the missing timestamp with the current timestamp.

Reply:
I don't know if this would be better handled in the UI with a default value. In Risk Management we fill the criticality based on the assumed impact, could we do a criticality calculation here based on date for example?

Reply:
> I don't know if this would be better handled in the UI with a default value. In Risk Management we fill the criticality based on the assumed impact, could we do a criticality calculation here based on date for example?

something like "if the date is more than a week in the past set the criticality to 'high'"?

Reply:
Yes, right now it's loaded from a CSV but we could check if Urgency is "high" and if it's older than a week then mark it as high in the criticality.